torch/_inductor/lowering.py

```diff
@@ -770,7 +770,7 @@ def select(x, dim, idx):
 def split(x, sizes, dim=0):
     dim = _validate_dim(x, dim, 0)
     x_size = V.graph.sizevars.guard_static_shape(x.get_size()[dim])
-    if isinstance(sizes, int):
+    if isinstance(sizes, (int, sympy.Integer)):
```

Comment:
```suggestion
    if isinstance(sizes, (int, sympy.Expr)):
```

Reply:
Hmm, I explicitly want an Integer handled here - because the rest seem nonsensical for this flow? Perhaps we can assert on if we see an Expr that is not an Integer?

Reply:
That seems reasonable in the short term.  I don't think it will do the correct thing currently.

Reply:
Well, I could also size_hint on it to get the size out, forcing a guard. It's up to you how you want it to behave.

Reply:
guard_static_shape is the inductor version of shape_env size_hint. For the future, maybe we want to dedupe some of the logic here...

Reply:
Why would expr be nonsensical? It's not too hard to imagine a code like
```
size = x.size(1)
torch.split(y, size//2)
```

Reply:
The presence of expr is not nonsensical, but the flow as its written today is. I can extract out a real value with guard_static_shape.